EVENTS:
  scene: "/home/alessandro/datasets/dvxplorer/day2_events/"
  # scene: "/home/alessandro/datasets/dvxplorer/night1_events/"
  side: "left"   #left 
  H: 480
  W: 640
  rectify: false
  use_valid_frame_range: true 



#CORNER DETECTOR PARAMETERS
FAST:
  threshold: 50
  nonmaxSuppression: false
  gradientScoring:
    enable: true
    desiredFeatures: 200

ACCUMULATOR:
  neutralPotential: 0.5
  eventContribution: 0.35
  decay: 1.0
  ignorePolarity: false

SLICING:
  type: "fixed"    # "fixed" | "adaptive"
  dt_ms: 33        # used only if type == "fixed"


CAMERA:
  #camera intrinsic parameters
  fx : 676.33153642211096
  fy : 677.78307833910333
  cx : 302.40954664576429
  cy : 250.16845570548003
  pixelSize: 0.000009 #9 uMeters
  gyroFreq: 800.0   #in Hz

  

LK:
  winSize:
    width: 40  # Width of the window for Lucas-Kanade optical flow  (default: 21)
    height: 40  # Height of the window for Lucas-Kanade optical flow  (default: 21)

  maxLevel: 4  # Maximum pyramid level for Lucas-Kanade optical flow (default: 3)

  criteria:
    maxCount: 30  # Maximum number of iterations for Lucas-Kanade optical flow (default: 30)
    epsilon: 0.01  # Threshold for termination criteria of Lucas-Kanade optical flow (default: 0.01)

  flags: 0  # Flags for Lucas-Kanade optical flow 
  minEigThreshold: 0.001  # Threshold for minimum eigenvalue of Lucas-Kanade optical flow 


#LOW PASS FILTER
SMOOTHINGFILTER:
  enable: true
  type: 0   # 0 : COMPLEMENTARY FILTER, 1 : LOW PASS FILTER
  lpfK: 0.7  #coefficient for the low pass filter. 0.5 is the default value
  complementaryK: 2.3 #coefficient for the complementary filter.

#altitude parameters
ALTITUDE:
  type: 0   # 0:average, 1:median
  saturationValue: 45   #in meters, if values over this are computed, than saturate to this

#safety parameters
lidarMax : 0.0    #in meters. if lidar the lidar detects less than this, publish this value rather than the vision based altitude

publishAltitudeFrequency: 50  #in Hz. IT MUST BE BETWEEN 30 and 50 Hz

velocitySource : 0 # 0: airspped, 1: groundspeed

FASTLKParallel : true



REJECTION_FILTER:
  magnitudeThresholdPixel: 10
  boundThreshold: 1.50          #threshold for rejecting the vectors magnitude. it must be > 1.0

#ADAPTIVE SLICING : 
adaptiveSlicing:
  enable: true
  P: 0.3
  I: 0.05
  D: 0.000
  maxTimingWindow: 50  #ms : min FPS = 20
  minTimingWindow: 10   #ms : max FPS = 100
  adaptiveTimingWindowStep: 1   #ms
  thresholdPIDEvents: 10
  OFPixelSetpoint: 5  #

#visualize image
visualizeImage: true
delayVisualize: 0